manifests/init.pp

```diff
@@ -482,6 +483,10 @@
     validate_re($sshd_config_strictmodes, '^(yes|no)$', "ssh::sshd_config_strictmodes may be either 'yes' or 'no' and is set to <${sshd_config_strictmodes}>.")
   }
 
+  if $sshd_hostbasedauthentication != undef {
```

Comment:
Since the default is a value and not undef, we don't need the conditional logic here.